### PR TITLE
feat: add ImportButton UI component

### PR DESCRIPTION
**Pull request by huynguyen-hl**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by ldhyen99 — APPROVED** [review]

File: packages/components/src/components/ImportButton/ImportButton.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import { Box } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { UploadFile as UploadFileIcon } from '@mui/icons-material';
+
+interface IProps {
+  buttonText?: string;
+  onChange: (data: object[]) => void;
+}
+
+export function loadJsonFile(file: File): Promise<object> {
```
Comment (namhoang1604) [inline packages/components/src/components/ImportButton/ImportButton.tsx:11]: Is there any reason for this function is outside the component `ImportButton`

File: packages/components/src/components/ImportButton/ImportButton.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    };
+
+    fileReader.readAsText(file);
+  });
+}
+
+/**
+ * ImportButton component is used to display the footer
+ */
+export const ImportButton = ({ buttonText = 'Import', onChange }: IProps) => {
```
Comment (namhoang1604) [inline packages/components/src/components/ImportButton/ImportButton.tsx:39]: `buttonText`, how about `label`?

File: packages/components/src/components/__tests__/ImportButton/ImportButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      // Expect the onChange function to have been called with the expected content
+      expect(fileContentObject).toMatchObject(contentObjectMock);
+    })} />);
+
+    const fileInput = screen.getByLabelText('Import');
+    fireEvent.change(fileInput, { target: { files: [fileMock] } });
+
+    // Wait for the loading state to resolve
+    await waitFor(() => expect(fileInput).not.toBeDisabled());
+  });
```
Comment (hoa-ngo-gs) [inline packages/components/src/components/__tests__/ImportButton/ImportButton.test.tsx:50]: Should we add a test case where an error occurs?

File: packages/components/src/components/ImportButton/ImportButton.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      >
+        <LoadingButton
+          loading={loading}
+          component='label'
+          variant='outlined'
+          startIcon={<UploadFileIcon />}
+          sx={{ margin: '0 5px' }}
+        >
+          {buttonText}
+          <input type='file' accept='.json' hidden onChange={handleFileUpload} multiple />
```
[the remark this answers is not in the document]
Comment (huynguyen-hl) [reply packages/components/src/components/ImportButton/ImportButton.tsx:86]: Currently, the component support only '.json' file extensions, but we plan to extend support in the future. Therefore, the component name doesn't need to change for general usage. Thank you for the reminder.

File: packages/components/src/components/ImportButton/ImportButton.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          display: 'flex',
+          alignItems: 'center',
+          marginBottom: '20px',
+        }}
+      >
+        <LoadingButton
+          loading={loading}
+          component='label'
+          variant='outlined'
+          startIcon={<UploadFileIcon />}
```
[the remark this answers is not in the document]
Comment (huynguyen-hl) [reply packages/components/src/components/ImportButton/ImportButton.tsx:82]: Yes, that would be a great feature for this component. However, we have limited time in this sprint, so I just keep this component simple and focus is on integrating all packages together. I'll plan to improve this component in the next sprint. Thank you for the advice.

[the remark this answers is not in the document]
Comment (huynguyen-hl) [reply packages/components/src/components/ImportButton/ImportButton.tsx:87]: Yes, that would be a great feature. We plan to support it in the future.

**Review by ldhyen99 — APPROVED** [review]

**Review by namhoang1604 — APPROVED** [review]

**Pull request merged**